sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -280,6 +280,29 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
   }
 
   /**
+   * Extracts the [[StructField]] with the given name recursively.
+   *
+   * @throws IllegalArgumentException if the parent field's type is not StructType
+   */
+  def getFieldRecursively(name: String): StructField = {
```

Comment:
I think I understood how it works. My point is, this is a Parquet-specific problem not related with Catalyst module. I don't see any reason that this method should be exposed.

I believe we can do this not by modifying the column names (not even for a temporary use).